conditional items in macro bind_interrupts!() not working anymore!

I am using conditional items in macro `bind_interrupts!()`.

I am using this for a while, was working +month ago. last embassy commit that is works was 27fb1f4d
But today I updated a older part of my project and now I am getting a compiler error.

Basic example, by editing `examples\stm32h5\src\bin\dts.rs`.
```
bind_interrupts!(struct Irqs {
    #[cfg(not(miri))]
    DTS => InterruptHandler;
});
```

Error:
```
<snip>\embassy\examples\stm32h5: cargo +1.85 build --release --target thumbv8m.main-none-eabihf
   Compiling embassy-stm32h5-examples v0.1.0 (<snip>\embassy\examples\stm32h5)
error: local ambiguity when calling macro `bind_interrupts`: multiple parsing options: built-in NTs meta ('inner') or 1 other option.
  --> src\bin\dts.rs:14:7
   |
14 |     #[cfg(not(miri))]
   |       ^^^

error[E0425]: cannot find value `Irqs` in this scope
  --> src\bin\dts.rs:55:35
   |
55 |     let mut dts = Dts::new(p.DTS, Irqs, config);
   |                                   ^^^^ not found in this scope

```

It seems that be20c708 is the issue.
If I go back 1 commit 870296fc the error is gone.

Real world example is multipule boards and featues like this
```
bind_interrupts!(struct Irqs {
    #[cfg(feature = "board_stm32h573idk")]
    ETH => hal::eth::InterruptHandler;
    #[cfg(not(feature = "board_stm32u083dk"))]
    RNG => hal::rng::InterruptHandler<peripherals::RNG>;
    #[cfg(feature = "board_stm32u083dk")]
    RNG_CRYP => hal::rng::InterruptHandler<peripherals::RNG>;

    #[cfg(all(feature = "board_spe", feature = "uart-dut-stlink"))]
    USART1 => usart::InterruptHandler<peripherals::USART1>;

    #[cfg(all(any(feature = "board_stm32h573idk", feature = "board_stm32u083dk"), feature =  "_usb"))]
    USB_DRD_FS => hal::usb::InterruptHandler<peripherals::USB>;

    #[cfg(all(feature = "board_stm32u083dk", feature = "_uart_mb"))]
    USART2_LPUART2 => usart::InterruptHandler<peripherals::USART2>;

    #[cfg(all(feature = "board_stm32h573idk", feature = "_uart_mb"))]
    UART7 => usart::InterruptHandler<peripherals::UART7>;
});
```

I am not sure if it is offical supported, because I could not find an example that used that feature.
I hope someone with macro knowlage knows how to fix it.

@Dirbaio @jake-taf

--- a/embassy-stm32/src/lib.rs
+++ b/embassy-stm32/src/lib.rs
@@ -178,7 +178,7 @@ pub use crate::_generated::interrupt;
 macro_rules! bind_interrupts {
     ($(#[$outer:meta])* $vis:vis struct $name:ident {
         $(
-            $(#[$inner:meta])*
+            $(#[doc = $doc:literal])*
             $(#[cfg($cond_irq:meta)])?
             $irq:ident => $(
                 $(#[cfg($cond_handler:meta)])?
@@ -194,7 +194,7 @@ macro_rules! bind_interrupts {
             #[allow(non_snake_case)]
             #[no_mangle]
             $(#[cfg($cond_irq)])?
-            $(#[$inner])*
+            $(#[doc = $doc])*
             unsafe extern "C" fn $irq() {
                 $(
                     $(#[cfg($cond_handler)])?

